### ISSUE TITLE
quarkus-kubernetes should support security-context

### ISSUE THREAD

**Issue by anthonydahanne** [quarkusio/quarkus#19385]
### Description

quarkus-kubernetes supports many features from the `Deployment` resource, such as : `image-pull-policy` , `liveness-probe`, etc., see https://quarkus.io/guides/deploying-to-kubernetes#configuration-options

Unfortunately, it does not support `securityContext` (see https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ for details about this feature)

### Implementation ideas

Not sure if the integration with fabric8 prevents adding this feature, it looks kind of big since there are many options to a `securityContext` 

I am using this workaround for now:

* I create a resource named `patch-deployment.yml` with this content:
    
```
spec:
  template:
    spec:
      containers:
        - name: my
          securityContext:
            runAsNonRoot: true
            runAsUser: 1001
```

* and then I patch my deployment:

```
kubectl  patch deployment my --patch "$(cat src/main/kubernetes/patch-deployment.yml)"
```

**Comment by geoand** [comment]
@iocanel does Dekorate have support for this?

**Comment by anthonydahanne** [comment]
Hello! Thanks for your answers.
Could you please give me some pointers, I could try to contribute the missing decorators? (Unless it's really trivial for you to add)
Thanks

**Comment by anthonydahanne** [comment]
ah, seems to be happening there: https://github.com/quarkusio/quarkus/blob/main/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ContainerConfig.java

**Comment by anthonydahanne** [comment]
and indeed dekorate does not support this feature as of today: https://github.com/dekorateio/dekorate/blob/master/assets/config.md

**Comment by iocanel** [comment]
This has been implemented as of d3a366b709e8f9dc53ce8b87c6642c6e0e018954
